torch/cuda/__init__.py

```diff
@@ -588,7 +590,11 @@ def set_stream(stream: Stream):
 
 def _parse_visible_devices() -> Union[List[int], List[str]]:
     r"""Parse CUDA_VISIBLE_DEVICES environment variable."""
-    var = os.getenv("CUDA_VISIBLE_DEVICES")
+    var = (
+        os.getenv("CUDA_VISIBLE_DEVICES")
+        if not torch.version.hip
+        else os.getenv("HIP_VISBILE_DEVICES")
```

Comment:
typo
```suggestion
        else os.getenv("HIP_VISIBLE_DEVICES")
```

Reply:
Could be more flexible to accept either CUDA or HIP env var.  Our HIP runtime also parses the CUDA_VISIBLE_DEVICES env var to ease porting.

```python
var = os.getenv("CUDA_VISIBLE_DEVICES") or os.getenv("HIP_VISIBLE_DEVICES")
```

If CUDA env var isn't set, it returns `None` and the `or` clause continues and checks the HIP env var.  If it returns `None`, `var` is finally set to None.